### PR TITLE
Add confirmation dialog when deleting a file.

### PR DESCRIPTION
**Pull request by daytonn**
# Display a confirmation dialog when deleting a resource.

This PR adds a confirmation dialog when attempting to delete a resource to prevent accidental deletion.
Peek 2020-08-14 10-07

# Checklist

- [x] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by kevin-inrupt** [comment]
Functionally this all works. The Confirm Delete could use a little TLC from a UX perspective.

**Comment by ajacksified** [comment]
The error message could also use a little love: you get `TypeError: Failed to fetch` when attempting to delete something that returns a non-200 response. (That's my fault from when I first implemented it - we should come back around to it, and not necessarily in this PR.)

File: components/resourceDetails/deleteLink/index.jsx [hunk 1]
```diff
@@ -19,30 +19,80 @@
...
+    useEffect(() => {
+      if (confirmationSetup && !confirmed) return;
 
-      try {
-        await deleteFile(resourceIri, { fetch: session.fetch });
-        onDelete();
-      } catch (err) {
-        onDeleteError(err);
+      setTitle("Confirm Delete");
+      setContent(<p>Are you sure you wish to delete this file?</p>);
```
Comment (ajacksified) [inline components/resourceDetails/deleteLink/index.jsx:69]: Replace "this file" with the filename?

File: components/resourceDetails/deleteLink/index.jsx [hunk 2]
```diff
@@ -19,30 +19,92 @@
...
+  setAlertOpen,
+  setMessage,
+  setSeverity,
+}) {
+  return async () => {
+    try {
+      await deleteFile(resourceIri, { fetch });
+      onDelete();
+      setSeverity("success");
+      setMessage(`Resource was successfully deleted.`);
```
Comment (ajacksified) [inline components/resourceDetails/deleteLink/index.jsx:42]: File name here too? 😁

**Pull request merged**